src/test/fuzz/util.h

```diff
@@ -54,4 +70,27 @@ NODISCARD inline CScriptNum ConsumeScriptNum(FuzzedDataProvider& fuzzed_data_pro
     return CScriptNum{fuzzed_data_provider.ConsumeIntegral<int64_t>()};
 }
 
+template <typename T>
+bool MultiplicationOverflow(T i, T j)
+{
+    static_assert(std::is_integral<T>::value, "Integral required.");
+    if (std::numeric_limits<T>::is_signed) {
+        if (i > 0) {
+            if (j > 0) {
+                return i > (std::numeric_limits<T>::max() / j);
+            } else {
+                return j < (std::numeric_limits<T>::min() / i);
+            }
+        } else {
+            if (j > 0) {
+                return i < (std::numeric_limits<T>::min() / j);
+            } else {
+                return i != 0 && (j < (std::numeric_limits<T>::max() / i));
+            }
+        }
+    } else {
+        return j != 0 && i > std::numeric_limits<T>::max() / j;
+    }
```

Comment:
Why implement this from scratch? Could just use `__builtin_mul_overflow`?

Reply:
I think it would be handy to have `MultiplicationOverflow(…)` in other non-test parts of the code base.

When we are confident of the equivalence between `MultiplicationOverflow` and `__builtin_mul_overflow` we can implement `MultiplicationOverflow` by using the `__builtin_mul_overflow` builtin where available (gcc + clang) and the current implementation where it is not (MSVC).

That's why I added a fuzz test for `MultiplicationOverflow` too.

Makes sense? :)

Reply:
When talking about integer overflows:

Don't miss this one: [a signed integer overflow in Microsoft's SafeInt ("SafeInt is a class library for C++ that manages integer overflows")](https://github.com/dcleblanc/SafeInt/issues/11)...

It's like rain on your wedding day
It's a free ride when you've already paid
It's the good advice that you just didn't take

:)